server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -37,13 +37,31 @@ public class EsExecutors {
     /**
      * Setting to manually control the number of allocated processors. This setting is used to adjust thread pool sizes per node. The
      * default value is {@link Runtime#availableProcessors()} but should be manually controlled if not all processors on the machine are
-     * available to Elasticsearch (e.g., because of CPU limits).
+     * available to Elasticsearch (e.g., because of CPU limits). Note that this setting accepts floating point processors, so it should not
+     * be used directly, use always {@link EsExecutors#allocatedProcessors(Settings)} to obtain a rounded number of processors.
      */
-    public static final Setting<Integer> NODE_PROCESSORS_SETTING = Setting.intSetting(
+    public static final Setting<Double> NODE_PROCESSORS_SETTING = new Setting<>(
         "node.processors",
-        Runtime.getRuntime().availableProcessors(),
-        1,
-        Runtime.getRuntime().availableProcessors(),
+        Double.toString(Runtime.getRuntime().availableProcessors()),
+        textValue -> {
+            double numberOfProcessors = Double.parseDouble(textValue);
+            if (Double.isNaN(numberOfProcessors) || Double.isInfinite(numberOfProcessors)) {
+                String err = "Failed to parse value [" + textValue + "] for setting [node.processors]";
+                throw new IllegalArgumentException(err);
+            }
+
+            if (numberOfProcessors < Double.MIN_VALUE) {
```

Comment:
Just curious, but is this different from `numberOfProcessors <= 0.0d`?

Reply:
Not really, I can change it to `numberOfProcessors <= 0.0d` if that's clearer